clusterloader/doc/vision.md

```diff
@@ -0,0 +1,369 @@
+be created in a given number of replicas in each of the namespace defined by namespace
+range and those creations should be spread over time using a referenced tuning set.
+
+
+#### PhaseUpdate
+
+PhaseUpdate defines a set of objects and a way how those should be updated within a
+single phase. We define it as following:
+
+```
+struct PhaseUpdate {
+	// Set of namespaces in which objects should be updated.
+	// If null, object is assumed to be cluster scope.
+	NamespaceRange Range
+	// Type of objects that should be updated.
+	ObjectType ObjectType
+	// Selector for objects to be updated within namespace.
+	Selector map[string]string
+	// Update function to be used to update an object.
+	UpdateFunction UpdateFunction
+	// Name of TunningSet to be used.
+	TuningSet string
+}
+```
+
+This describes that objects of a given type from a given namespaces selected by a
+given selector should be updated with a given update function and those updates
+should be spread over time using a referenced tuning set.
+
+What is still missing is defining how to do an update. As a starting point, we allow
+for two types of updates: scaling the resource (that is possible only for some
+resources exposing /scale subresource) and changing some field to given values. We
+define it as following:
+
+```
+struct UpdateFunction {
+	// Exactly one of the fields should be set.
+
+	// Scale resource using its /scale subresource
+	Scale *ScaleParameters
+	// Change a set of fields for the object to given values.
+	UpdateFields *UpdateFieldsParameters
+}
+
+struct ScaleParameters {
+	// Exactly one of the fields should be set.
+
+	// Scale to a given value.
+	DesiredReplicas *int32
+	// Scale to some random value in a given range.
+	RandomReplicas *Range
+}
+
+struct UpdateFieldsParameters {
+	UpdateFields []UpdateField
+}
+
+struct UpdateField {
+	// Path a a field that should be modified (in the same format as
+	// fields are defined in Kubernetes field selectors).
+	Field string
+	// Desired value of the field (json-serialized for non-string fields)
+	Value string
+}
+```
+
+In there will be a need to support other types of updates in the future, it should
+be fairly simple to extend these types.
+
+
+#### PhaseDelete
+
+PhaseDelete defines a set of objects that should be deleted within a single phase.
+We define it as following:
+
+```
+struct PhaseDelete {
+	// Set of namespaces in which objects should be updated.
+	// If null, object is assumed to be cluster scope.
+	NamespaceRange Range
+	// Type of objects that should be updated.
+	ObjectType ObjectType
+	// Selector for objects to be updated within namespace.
+	Selector map[string]string
+	// Name of TunningSet to be used.
+	TuningSet string
+}
+```
+
+This describes that objects of a given type from a given namespaces selected by a
+given selector should be deleted and those updates should be spread over time using
+a referenced tuning set.
+
+
+### Measurements
```

Comment:
I like building in the measurement capture into Cluster Loader..

Reply:
Thanks!